Add lolcat feature

nswarup14 opened the pull request:
The animation feature is not working. I've marked the line where the error seems to occur as a comment

nswarup14 committed:
Add lolcat feature

nswarup14 committed:
Remove the feature of animation

nswarup14 committed:
Move the lolcat class to a new python script. Update co-author and version number

naveen-u commented on todo:
```diff
@@ -46,21 +48,27 @@ def printTodo(todo, ids, verbose, searchTags):
 				continue
 			if i[0] == 1:
-				s = '    \u2757 \u2B50\t'
+				s = '\u2757 \u2B50\t'
```
```suggestion
				s = '\t\u2757 \u2B50\t'
```

naveen-u commented on todo:
```diff
@@ -46,21 +48,27 @@ def printTodo(todo, ids, verbose, searchTags):
+				s = '\u2757 \u2B50\t'
 			elif i[0] == 2:
-				s = '    \u2757\t'
+				s = '\u2757\t'
```
```suggestion
				s = '\t\u2757\t'
```

naveen-u commented on todo:
```diff
@@ -46,21 +48,27 @@ def printTodo(todo, ids, verbose, searchTags):
+				s = '\u2757\t'
 			elif i[0] == 3:
-				s = '      \u2B50\t'
+				s = '\u2B50\t'
```
```suggestion
				s = '\t\u2B50\t'
```

naveen-u commented on lolcat.py:
```diff
@@ -0,0 +1,124 @@
+        	rgb = self.rainbow(options.freq, options.seed + i / options.spread)
+        	self.output.write(''.join([self.wrap(self.ansi(rgb)), c if PY3 else c.encode(options.charset_py2, 'replace')]))
+
+def detect_mode(term_hint='xterm-256color'):
+    '''
+    Poor-mans color mode detection.
+    '''
+    if 'ANSICON' in os.environ:
+        return 16
+    elif os.environ.get('ConEmuANSI', 'OFF') == 'ON':
+        return 256
+    else:
+        term = os.environ.get('TERM', term_hint)
+        if term.endswith('-256color') or term in ('xterm', 'screen'):
+            return 256
+        elif term.endswith('-color') or term in ('rxvt',):
+            return 16
+        else:
+            return 256 # optimistic default
```
Put this method inside `Lolcat` and initialise `Lolcat` settings (`spread`, `mode`, and `seed`) from inside `cat`.

nswarup14 replied:
Done @naveen-u

nswarup14 replied:
Refactored the code accordingly. Moved the `method` inside the `LolCat` class

naveen-u commented on todo:
```diff
@@ -210,4 +228,11 @@ if __name__ == '__main__':
-		printTodo(todo, ids, args.verbose, args.tag_main)
\ No newline at end of file
+		if args.command == 'lolcat':
+			if args.spread == 0.0:
```
We should check for `<=0` instead of `==0`.

naveen-u commented on todo:
```diff
@@ -210,4 +228,11 @@ if __name__ == '__main__':
+			if args.spread == 0.0:
+				args.spread = 3.0
+			args.mode = args.mode or detect_mode()
+			args.seed = random.randint(0, 256) if args.seed == 0 else args.seed
```
Here too, check for `<=0` instead of `==0`.

naveen-u commented on todo:
```diff
@@ -210,4 +228,11 @@ if __name__ == '__main__':
 	if args.suppress is None or args.suppress is False:
-		printTodo(todo, ids, args.verbose, args.tag_main)
\ No newline at end of file
+		if args.command == 'lolcat':
+			if args.spread == 0.0:
+				args.spread = 3.0
+			args.mode = args.mode or detect_mode()
+			args.seed = random.randint(0, 256) if args.seed == 0 else args.seed
+			printTodo(todo, ids, args.verbose, args.tag_main, args, enable_lolcat=True)
+		else:
+			printTodo(todo, ids, args.verbose, args.tag_main, args)
```
1. Would be cleaner to just call `printTodo(todo, ids, args, enable_lolcat=(args.command == 'lolcat'))` here. You should check and initialise `Lolcat` settings (`spread`, `mode`, and `seed`) inside the `Lolcat` class. `detect_mode` can be made a method of `Lolcat` as it won't be used anywhere else.

2. Check for `<=0` instead of `==0` when setting `spread` and `seed`.

3. Passing `args.verbose` and `args.tag_main` are redundant as you're passing `args` as a whole. Refactor the `printTodo` method to just take `args`.

nswarup14 replied:
Refactored the code accordingly

naveen-u reviewed (changes requested):
Minor refactoring required.

nswarup14 committed:
Refactor LolCat class and remove support for PY2

1. Initialise various arguments within LolCat class
2. Rename the 'del' subparser
3. Remove redundant arguments being passed

nswarup14 committed:
Adding a tab before printing todo entry

nswarup14 commented:
@naveen-u 
I refactored what you had requested. The code looks a lot cleaner now tbh.
But, I was playing around with `git rebase` because I had a `Merge Commit` and thus, might have rewritten some the of the commit history a bit differently. But the commit messages are correct and the script works perfectly, just the ordering has been changed.

Let me know what you think
